Sources/PubNub/Helpers/Crypto/CryptorModule.swift

```diff
@@ -101,45 +82,58 @@ public struct CryptorModule {
   public func decrypt(data: Data) -> Result<Data, PubNubError> {
     guard !data.isEmpty else {
       return .failure(PubNubError(
-        .decryptionError,
-        additional: ["Cannot decrypt empty Data"])
+        .decryptionFailure,
+        additional: ["Cannot decrypt empty Data in \(String(describing: self))"])
       )
     }
     do {
       let header = try CryptorHeader.from(data: data)
       
       guard let cryptor = cryptor(matching: header) else {
         return .failure(PubNubError(
-          .unknownCryptorError,
+          .unknownCryptorFailure,
           additional: [
             "Could not find matching Cryptor for \(header.cryptorId()) while decrypting Data. " +
             "Ensure the corresponding instance is registered in \(String(describing: Self.self))"
           ]
         ))
       }
+      
+      let metadata: Data
+      let contentData: Data
+      
+      switch header {
+      case .none:
+        metadata = Data()
+        contentData = data
```

Comment:
The lack of metadata can be also represented by empty `Data` object. Also, this class provides [isEmpty](https://developer.apple.com/documentation/foundation/data/1780284-isempty) method. I wanted to avoid optional variable that always requires you to write more code or perform additional checks.

Reply:
OK, makes sense ;)